internal/cli/cmd/promote/promote.go

```diff
@@ -0,0 +1,129 @@
+package promote
+
+import (
+	"fmt"
+
+	"connectrpc.com/connect"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/utils/ptr"
+
+	typesv1alpha1 "github.com/akuity/kargo/internal/api/types/v1alpha1"
+	"github.com/akuity/kargo/internal/cli/client"
+	"github.com/akuity/kargo/internal/cli/config"
+	"github.com/akuity/kargo/internal/cli/option"
+	v1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
+)
+
+func NewCommand(cfg config.CLIConfig, opt *option.Option) *cobra.Command {
+	var freight string
+	var stage string
+	var subscribersOf string
+
+	cmd := &cobra.Command{
+		Use:   "promote [--project=project] --freight=freight-id [--stage=stage] [--subscribers-of=stage]",
+		Short: "Manage the promotion of freight",
+		Args:  option.NoArgs,
+		Example: `
+# Promote a freight to a stage for a specific project
+kargo promote --project=my-project --freight=abc123 --stage=dev
+
+# Promote a freight to subscribers of a stage for a specific project
+kargo promote --project=my-project --freight=abc123 --subscribers-of=dev
+
+# Promote a freight to a stage for the default project
+kargo config set project my-project
+kargo promote --freight=abc123 --stage=dev
+
+# Promote a freight to subscribers of a stage for the default project
+kargo config set project my-project
+kargo promote --freight=abc123 --subscribers-of=dev
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+
+			project := opt.Project
+			if project == "" {
+				return errors.New("project is required")
+			}
+
+			if freight == "" {
+				// TODO: Get latest available freight if empty
```

Comment:
I know this comment moved here from another file, but this TODO can go away. Scenarios where we can safely infer the desired freight are few and far between. It is better that we leave the flag required.